chainntnfs/bitcoindnotify/bitcoind.go

```diff
@@ -785,32 +820,64 @@ func (b *BitcoindNotifier) historicalSpendDetails(
 	spendRequest chainntnfs.SpendRequest, startHeight, endHeight uint32) (
 	*chainntnfs.SpendDetail, error) {
 
-	// Begin scanning blocks at every height to determine if the outpoint
-	// was spent.
+	// batchRequest will store the scan requests at every height to determine
+	// if the output was spent.
+	batchRequest := make(
+		map[uint32]rpcclient.FutureGetBlockHashResult, endHeight-startHeight,
+	)
+
 	for height := endHeight; height >= startHeight && height > 0; height-- {
+		// BatchClient works with AsynRequest.
+		sendReq, err := b.chainConn.GetBlockHashAsync(int64(height))
+		if err != nil {
+			return nil, fmt.Errorf("unable to receive the block promise %v", err)
+		}
+		batchRequest[height] = sendReq
+	}
+
+	// Sending the bulk request using updated batchClient.
+	b.chainConn.SendAsyncQueue()
+
+	blockHashAddresses := make([]*chainhash.Hash, 0, len(batchRequest))
+
+	for height, response := range batchRequest {
 		// Ensure we haven't been requested to shut down before
-		// processing the next height.
+		// processing the next height response.
 		select {
 		case <-b.quit:
 			return nil, chainntnfs.ErrChainNotifierShuttingDown
 		default:
 		}
 
 		// First, we'll fetch the block for the current height.
-		blockHash, err := b.chainConn.GetBlockHash(int64(height))
+		blockHash, err := response.Receive()
 		if err != nil {
-			return nil, fmt.Errorf("unable to retrieve hash for "+
-				"block with height %d: %v", height, err)
+			return nil, fmt.Errorf("unable to retrieve hash for block with "+
+				"height %d: %v", height, err)
 		}
-		block, err := b.GetBlock(blockHash)
-		if err != nil {
-			return nil, fmt.Errorf("unable to retrieve block "+
-				"with hash %v: %v", blockHash, err)
+
+		blockHashAddresses = append(blockHashAddresses, blockHash)
+	}
+
+	blocks, err := b.chainConn.GetBlocksBatch(blockHashAddresses)
```

Comment:
Similar comment to the above that we should slice this up a little to not fetch more than 10-25 blocks at once.